src/platform/stm32f1/spi.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * SPDX-License-Identifier: MIT
+ * SPDX-FileCopyrightText: 2021 Rafael Silva <silvagracarafael@gmail.com>
+ */
+
+#pragma once
+
+#include <stm32f1xx.h>
+
+#include "platform/stm32f1/gpio.h"
+
+#define SPI_MSB_FIRST 0
+#define SPI_LSB_FIRST 1
+
+#define SPI_MODE0 0
+#define SPI_MODE1 1
+#define SPI_MODE2 2
+#define SPI_MODE3 3
```

Comment:
Maybe make this an enum?

Reply:
why?

Reply:
Better type checking than enums. This ensures you can only pass valid values to `spi_init`. eg. prevents people from mistakenly setting the mode to 11.

Reply:
makes sense, ok